fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt

```diff
@@ -149,7 +150,11 @@ class DefaultToolbarIntegration(
 
         tabsAction.updateCount(tabCount)
 
-        toolbar.addBrowserAction(tabsAction)
+        if (showTabCounter) {
+            toolbar.addBrowserAction(tabsAction)
+        } else {
+            toolbar.removeBrowserAction(tabsAction)
```

Comment:
It looks like we are removing an action here that has never been added in the first place :) 

There is a bunch of code related to the tab menu starting from [here](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/ToolbarIntegration.kt#119) that doesn't make sense to run if the tab counter shouldn't be displayed.